addons/escoria-core/game/core-scripts/esc/commands/change_scene.gd

```diff
@@ -92,15 +92,18 @@ func run(command_params: Array) -> int:
 			var script = escoria.esc_compiler.load_esc_file(
 				room_scene.esc_script
 			)
-		
+			
 			if script.events.has("setup"):
 				escoria.event_manager.queue_event(script.events["setup"])
 				var rc = yield(escoria.event_manager, "event_finished")
 				while rc[1] != "setup":
 					rc = yield(escoria.event_manager, "event_finished")
 				if rc[0] != ESCExecution.RC_OK:
 					return rc[0]
-
+			
```

Comment:
That is because the :ready events are started before the transition is finished, so the player misses the beginning of the actions because they are hidden behind the transition.

Starting the transition (and waiting for it to finish) before :ready starts avoids the problem.

Reply:
Maybe we should add examples to the [event documentation](https://github.com/godotengine/escoria/blob/main/docs/esc.md#events) about :setup and :ready. Based on your description I don't get why ready should only be called the first time the player visits a scene.

Reply:
Based on the conversation in the comments, this can be resolved now.